src/analysis/seminorm.lean

```diff
@@ -0,0 +1,212 @@
+/-
+Copyright (c) 2019 Jean Lo. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jean Lo
+-/
+
+import algebra.pointwise
+import analysis.normed_space.basic
+
+/-!
+# Seminorms and local convexity
+
+This file introduces the following notions, defined for a vector space
+over a normed field:
+
+- the subset properties of being `absorbent` and `balanced`,
+
+- a `seminorm`, a function to the reals that is positive-semidefinite,
+  absolutely homogeneous, and subadditive.
+
+We prove related properties.
+
+## TODO
+
+Define and show equivalence of two notions of local convexity
+for a t.v.s. over ℝ or ℂ: that it has a local base of balanced convex
+absorbent sets, and that it carries the initial topology induced by a
+family of seminorms.
+
+## References
+* [H. H. Schaefer, *Topological Vector Spaces*][schaefer1966]
+-/
+
+-- subset properties : absorbent and balanced sets in a vector space
```

Comment:
You can use comments that show up in the doc with `/-! ... -/`, and you can even add markdown with titles. This would be great here, and also before the part on seminorms.